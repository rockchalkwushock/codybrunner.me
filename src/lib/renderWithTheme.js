import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from './theme'

const renderWithTheme = Component =>
  mount(<ThemeProvider theme={theme}>{Component}</ThemeProvider>)

export default renderWithTheme
