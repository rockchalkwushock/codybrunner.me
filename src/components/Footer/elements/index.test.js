import React from 'react'

import renderWithTheme from '../../../lib/renderWithTheme'
import { Container, Footer } from './'

describe('Component: <Container />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = renderWithTheme(<Container />)
  })
  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('should have styles', () => {
    expect(wrapper.children().props()).toHaveProperty('left', undefined)
    expect(wrapper).toHaveStyleRule('align-items', 'center')
    expect(wrapper).toHaveStyleRule('display', 'flex')
    expect(wrapper).toHaveStyleRule('flex-direction', 'column')
    expect(wrapper).toHaveStyleRule('justify-content', 'center')
    expect(wrapper).toHaveStyleRule('text-align', 'center')
    expect(wrapper).toHaveStyleRule('align-items', 'right', {
      media: '(min-width:736px)'
    })
  })
  test('should have different style with prop', () => {
    wrapper = renderWithTheme(<Container left />)
    expect(wrapper.children().props()).toHaveProperty('left', true)
    expect(wrapper).toHaveStyleRule('align-items', 'left', {
      media: '(min-width:736px)'
    })
  })
})

describe('Component: <Footer />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = renderWithTheme(<Footer />)
  })
  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('should have styles', () => {
    expect(wrapper).toHaveStyleRule('align-items', 'center')
    expect(wrapper).toHaveStyleRule('display', 'flex')
    expect(wrapper).toHaveStyleRule('flex-direction', 'column')
    expect(wrapper).toHaveStyleRule('font-size', '80%')
    expect(wrapper).toHaveStyleRule('justify-content', 'center')
    expect(wrapper).toHaveStyleRule('padding', '1em')
    expect(wrapper).toHaveStyleRule('flex-direction', 'row', {
      media: '(min-width:736px)'
    })
    expect(wrapper).toHaveStyleRule('font-size', '100%', {
      media: '(min-width:736px)'
    })
    expect(wrapper).toHaveStyleRule('justify-content', 'space-between', {
      media: '(min-width:736px)'
    })
  })
})
