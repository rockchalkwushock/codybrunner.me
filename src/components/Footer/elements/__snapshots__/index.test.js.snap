// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Container /> should render correctly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
}

@media (min-width:736px) {
  .c0 {
    -webkit-align-items: right;
    -webkit-box-align: right;
    -ms-flex-align: right;
    align-items: right;
  }
}

<ThemeProvider
  theme={
    Object {
      "post": Object {
        "bg": "#f9f8eb",
        "ftClr": "#000",
        "ftFm": "\\"Raleway\\", sans-serif",
        "ftSz": "1.1rem",
        "ftWt": "400",
        "linkClr": "#143a52",
        "linkHvClr": "#7a9eb1",
        "linkWt": "600",
        "lnHt": "1.618",
        "ltSp": "1px",
      },
      "screen": Object {
        "large": "736px",
      },
      "site": Object {
        "bg": "#7a9eb1",
        "borderRadius": "10px",
        "fg": "#ffe1b6",
        "ftClr": "#143a52",
        "ftFm": "\\"Raleway\\", sans-serif",
        "ftSz": "1rem",
        "ftWt": "400",
        "linkClr": "#143a52",
        "linkHvClr": "#ffe1b6",
        "lnHt": "1.4",
        "ltSp": "0px",
        "shadow": "1px 1px 5px 1px rgba(0, 0, 0, 0.7)",
        "transition": "all 0.3s ease-in-out",
      },
    }
  }
>
  <Container>
    <div
      className="c0"
    />
  </Container>
</ThemeProvider>
`;

exports[`Component: <Footer /> should render correctly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 80%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1em;
}

@media (min-width:736px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    font-size: 100%;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

<ThemeProvider
  theme={
    Object {
      "post": Object {
        "bg": "#f9f8eb",
        "ftClr": "#000",
        "ftFm": "\\"Raleway\\", sans-serif",
        "ftSz": "1.1rem",
        "ftWt": "400",
        "linkClr": "#143a52",
        "linkHvClr": "#7a9eb1",
        "linkWt": "600",
        "lnHt": "1.618",
        "ltSp": "1px",
      },
      "screen": Object {
        "large": "736px",
      },
      "site": Object {
        "bg": "#7a9eb1",
        "borderRadius": "10px",
        "fg": "#ffe1b6",
        "ftClr": "#143a52",
        "ftFm": "\\"Raleway\\", sans-serif",
        "ftSz": "1rem",
        "ftWt": "400",
        "linkClr": "#143a52",
        "linkHvClr": "#ffe1b6",
        "lnHt": "1.4",
        "ltSp": "0px",
        "shadow": "1px 1px 5px 1px rgba(0, 0, 0, 0.7)",
        "transition": "all 0.3s ease-in-out",
      },
    }
  }
>
  <Footer>
    <footer
      className="c0"
    />
  </Footer>
</ThemeProvider>
`;
